py3status/modules/updates.py

```diff
@@ -0,0 +1,215 @@
+            return None
+
+    def count_updates(self, output):
+        return len(output.splitlines())
+
+    def get_updates(self):
+        output = self.get_output()
+        if output is None:
+            return {self.name: output}
+        return {self.name: self.count_updates(output)}
+
+
+class Apt(Update):
+    def count_updates(self, output):
+        return len(output.splitlines()[1:])
+
+
+class Apk(Update):
+    def count_updates(self, output):
+        return len(output.splitlines()[1:])
+
+
+class Eopkg(Update):
+    def get_output(self):
+        output = self.parent.py3.command_output(self.command)
+        if "No packages to upgrade." in output:
+            return ""
+        return output
+
+
+class Zypper(Update):
+    def count_updates(self, output):
+        return len([x for x in output.splitlines() if x][4:])
+
+
+class Py3status:
+    """
+    """
+
+    # available configuration parameters
+    cache_timeout = 600
+    format = None
+    thresholds = [
+        (0, 'darkgray'), (10, 'degraded'), (20, 'orange'), (30, 'bad')]
+
+    class Meta:
+        deprecated = {
+            "rename_placeholder": [
+                {
+                    "placeholder": "aur",
+                    "new": "auracle",
+                    "format_strings": ["format"],
+                }
+            ]
+        }
+
+    def post_config_hook(self):
+        managers = update_managers = [
+            ("Pacman", "checkupdates"),
+            ("Auracle", "auracle sync --color=never"),
+            ("Yay", "yay --query --upgrades --aur"),
+            ("Apk", "apk version -l '<'"),
+            ("Apt", "apt list --upgradeable"),
+            ("Eopkg", "eopkg list-upgrades"),
+            ("Pkg", "pkg upgrade --dry-run --quiet"),
+            ("Xbps", "xbps-install --update --dry-run"),
+            ("Zypper", "zypper list-updates"),
+        ]
+
+        managed = getattr(self, 'managers', None)
+        if managed:
+            new_managers = []
+            for custom in managed:
+                if isinstance(custom, tuple):
+                    if len(custom) != 2 or custom[0].lower() == 'update':
+                        raise Exception(STRING_INVALID)
+                    new_managers.append(custom)
+                else:
+                    name = custom.lower()
+                    if name == 'update':
+                        raise Exception(STRING_INVALID)
+                    for manager in update_managers:
+                        if manager[0].lower() == name:
+                            new_managers.append(manager)
+                            break
+            managers = new_managers
+
+        placeholders = self.py3.get_placeholders_list(self.format or '')
+        placeholders = [x for x in placeholders if x != 'update']
+
+        formats = []
+        self.backends = []
+        for name, command in managers:
+            name_lowercased = name.lower()
+            if placeholders:
```

Comment:
try to avoid having `if`, ensure `placeholders` is an empty list so that `for` loop can take care of it

Reply:
This is a gibberish. I need to know placeholders is not empty so I can start using `break/continue`.   Otherwise, I keep going to the next step which is a binary check. This allow users to ignore installed `yay` if we have just `{pacman}` in the format.

Reply:
I think I finally got the logic, let's use something more readable

instead of

```python
if placeholders:
    for placeholder in placeholders:
        if placeholder == name_lowercased:
            break
        else:
            continue
```

How about:

```python
if len(placeholders) > 0 and name_lowercased not in placeholders:
    continue
```